forem/forem
# First version of exporting the Users to a CSV

## What type of PR is this? (check all applicable)


## Description
This is the first version of being able to export the Users table to a CSV. Whilst it is not yet perfect, we have the correct fields exporting. It does not implement the modal as per the design and that will be done in a subsequent PR.

I will highlight the improvements that I will make in subsequent PR's.
## Related Tickets & Documents



- Related Issue #https://github.com/forem/forem/issues/16934

## QA Instructions, Screenshots, Recordings

- Go to /admin/users. 
- Click on the export button
- Open the CSV that downloads and check that it matches the table. 

Demo can be found here: 
https://www.loom.com/share/fdda3c5027a848f3b267a37786e41976

### UI accessibility concerns?
Right now the button downloads the CSV directly but in a subsequent PR the button will open a modal and so I did not focus on accessibility since the role of the button will change.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
